odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/service/DataQualityServiceImpl.java

```diff
@@ -8,39 +9,44 @@
 import org.opendatadiscovery.oddplatform.exception.NotFoundException;
 import org.opendatadiscovery.oddplatform.mapper.DataEntityMapper;
 import org.opendatadiscovery.oddplatform.mapper.DataQualityMapper;
-import org.opendatadiscovery.oddplatform.repository.DataQualityRepository;
+import org.opendatadiscovery.oddplatform.repository.DataEntityRepository;
+import org.opendatadiscovery.oddplatform.repository.reactive.ReactiveDataQualityRepository;
 import org.springframework.stereotype.Service;
 import reactor.core.publisher.Mono;
 
 @Service
 @RequiredArgsConstructor
 public class DataQualityServiceImpl implements DataQualityService {
-    private final DataQualityRepository dataQualityRepository;
+    private final ReactiveDataQualityRepository dataQualityRepository;
+    private final DataEntityRepository dataEntityRepository;
     private final DataQualityMapper dataQualityMapper;
     private final DataEntityMapper dataEntityMapper;
 
     @Override
-    public Mono<DataEntityList> getDataEntityDataQATests(final long dataEntityId) {
-        return Mono.fromCallable(() -> dataQualityRepository.getDataQualityTests(dataEntityId))
+    public Mono<DataEntityList> getDatasetTests(final long datasetId) {
+        return dataQualityRepository.getDataQualityTestOddrnsForDataset(datasetId)
+            .switchIfEmpty(Mono.error(new NotFoundException("Dataset with id %d not found".formatted(datasetId))))
```

Comment:
"Data quality tests for dataset with id %d not found"

Reply:
Fixed